mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobClusterActor.java

```diff
@@ -1895,6 +1908,55 @@ private long getExpireAcceptedDelayMs() {
         return empty();
     }
 
+
+    private Optional<JobDefinition> createJobDefinitionInheritingInstanceCount(
+            final JobDefinition givenJobDefn,
+            final List<JobInfo> existingJobsList,
+            final List<CompletedJob> completedJobs,
+            final MantisJobStore store) {
+        if(logger.isTraceEnabled()) { logger.trace("Enter createJobDefinitionInheritingInstanceCount"); }
+
+        // use last submitted job (if present) to merge with the givenJobDefn so the result jobDefn will inherit instance count.
+        JobDefinition.Builder jobDefnBuilder = new JobDefinition.Builder().from(givenJobDefn);
+        Optional<JobDefinition> lastSubmittedJobDefn = getLastSubmittedJobDefinition(existingJobsList, completedJobs, empty(), store);
+        if (lastSubmittedJobDefn.isPresent()) {
+            if (givenJobDefn.getSchedulingInfo() == null) {
+                logger.error("Empty scheduling info in given job defn {}.", givenJobDefn);
+                return empty();
+            }
+
+            if (lastSubmittedJobDefn.get().getSchedulingInfo() != null &&
+                    lastSubmittedJobDefn.get().getSchedulingInfo().getStages().size() == givenJobDefn.getSchedulingInfo().getStages().size()) {
+                // Use instance count from last submitted job at each stage.
+                Map<Integer, StageSchedulingInfo> givenStages = givenJobDefn.getSchedulingInfo().getStages();
+                Map<Integer, StageSchedulingInfo> lastStages = lastSubmittedJobDefn.get().getSchedulingInfo().getStages();
+                SchedulingInfo.Builder siBuilder = new SchedulingInfo.Builder().numberOfStages(givenStages.size());
+                givenStages.forEach((stageId, givenStageInfo) -> {
+                    if (lastStages.containsKey(stageId) && givenStageInfo.getInheritInstanceCount()) {
+                        StageSchedulingInfo mergedStageInfo = new StageSchedulingInfo.Builder()
+                                .setNumberOfInstances(lastStages.get(stageId).getNumberOfInstances())
```

Comment:
Unfortunately, this doesn't give the current worker count for a stage. This only gives the worker count in job definition when the job was last submitted. The actual current worker count is dynamically changing from auto scaling. I think you would need the jobActor to get the up to date worker count.

Reply:
@calvin681 Updated the actor logic and test. Could you take a look when you have a chance?